index/index.ml

```diff
@@ -55,24 +74,50 @@ let main
       close_in h ;
       files @ other_files
   in
+  let files = List.map Fpath.v files in
+  let favourite_files = List.map Fpath.v favourite_files in
+  let odocl_files = List.filter (Fpath.has_ext "odocl") files in
+  let odoc_index_files = List.filter (Fpath.has_ext "odoc-index") files in
+  let favourite_odocl_files = List.filter (Fpath.has_ext "odocl") favourite_files in
+  let favourite_odoc_index_files =
+    List.filter (Fpath.has_ext "odoc-index") favourite_files
+  in
+  check_files_extension files ;
   let h = Storage.open_out db_filename in
   let flush () =
     let t = Db_writer.export ~summarize:(db_format = `ancient) db in
     Storage.save ~db:h t
   in
-  let loop ~favourite odoc =
+  let loop kind ~favourite file =
+    let odoc = Fpath.to_string file in
     let pkg, odoc =
       match String.split_on_char '\t' odoc with
```

Comment:
It looks very weird that we are parsing a tab-separated format from the `odoc` variable that was previously treated as a valid filename... Maybe it would be easier to check if the filename is an `.odocl` or `.odoc-index` on line 71 below, instead of doing a bunch of `List.filter` above? :)